File: src/services/authService.js
import api from "../api/api";

export const getCaptcha = () => api.get("/api/auth/captcha");

export const login = (payload) => api.post("/api/auth/login", payload);

export const signup = (payload) => api.post("/api/auth/register", payload);

export const verifyOtp = (payload) => api.post("/api/auth/verify-otp", payload);

export const resendOtp = (payload) => api.post("/api/auth/resend-otp", payload);

export const verifyEmail = (token) =>
  api.get("/api/auth/verify", { params: { token } });

export const forgotPassword = (payload) => api.post("/api/auth/forgot-password", payload);

export const resetPassword = (payload) => api.post("/api/auth/reset-password", payload);

export const logoutApi = (refreshToken) =>
  api.post("/api/auth/logout", { refreshToken });

File: backend/src/main/java/com/cdac/application_management_system/service/AuthService.java
package com.cdac.application_management_system.service;

import com.cdac.application_management_system.dto.request.*;
import com.cdac.application_management_system.dto.response.AuthResponseDTO;
import com.cdac.application_management_system.dto.response.CaptchaResponseDTO;
import com.cdac.application_management_system.dto.response.RegisterResponseDTO;

public interface AuthService {

    CaptchaResponseDTO getCaptcha();

    RegisterResponseDTO register(RegisterRequestDTO request);

    AuthResponseDTO login(LoginRequestDTO request);

    void verifyEmail(String token);

    void verifyOtp(VerifyOtpRequestDTO request);

    String resendOtp(ResendOtpRequestDTO request);

    void forgotPassword(ForgotPasswordRequestDTO request);

    void resetPassword(ResetPasswordRequestDTO request);

    AuthResponseDTO refresh(String refreshTokenValue);

    void logout(String refreshTokenValue);
}
